packages/pigeon/lib/generator_tools.dart

```diff
@@ -351,13 +352,23 @@ Map<TypeDeclaration, List<int>> getReferencedTypes(
   return references.map;
 }
 
+bool _canContainAnyType(TypeDeclaration type) {
+  return (type.baseName == 'List' && type.typeArguments.isEmpty) ||
```

Comment:
What about `List<Object>` and `Map<Object, Object>` (or maybe even `Map<anything, Object>`)?

Reply:
The recursive function `getReferencedTypes` will handle that.